map.go

```diff
@@ -2,11 +2,31 @@ package lo
 
 // Keys creates an array of the map keys.
 // Play: https://go.dev/play/p/Uu11fHASqrU
-func Keys[K comparable, V any](in map[K]V) []K {
-	result := make([]K, 0, len(in))
+func Keys[K comparable, V any](in ...map[K]V) []K {
+	result := make([]K, 0)
 
-	for k := range in {
-		result = append(result, k)
+	for i := range in {
+		for k := range in[i] {
+			result = append(result, k)
+		}
+	}
+
+	return result
+}
+
+// UniqKeys creates an array of unique keys in the map.
+// Play: https://go.dev/play/p/RyIE6Pb5dVS
+func UniqKeys[K comparable, V any](in ...map[K]V) []K {
+	seen := make(map[K]struct{})
+	result := make([]K, 0)
+
+	for i := range in {
+		for k := range in[i] {
+			if _, exists := seen[k]; !exists {
+				seen[k] = struct{}{}
+				result = append(result, k)
```

Comment:
```suggestion
			_, exists := seen[k]
			if exists {
				continue
			}
			seen[k] = struct{}{}
			result = append(result, k)
```

Reply:
It's the logic of "leave early", it's for readability and maintainability.

First, here, it's simpler to read that when existing we skip, than reading than when no existing we have to do something.

The leave early concept is there to avoid big if branch.

Also, imagine you have an new condition in this loop

With your current code, it would lead to put an if in the if

So the code would be something like this

```
for {
    if condition1 {
         something1
         if condition2 {
               something2 
         }
    }
}
```

Here is the same code when applying leave early

```
for {
    if !condition1 {
         continue
    }
    something1
    if !condition2 {
          continue
    }
    something2 
}
```

Reply:
waaaooo this seems a very good explanation. Thank you making it soo clear